x/incentives/simulation/operations.go

```diff
@@ -62,26 +63,42 @@ func WeightedOperations(
 	}
 }
 
+// genRewardCoins generates a random number of coin denoms with a respective random value for each coin.
 func genRewardCoins(r *rand.Rand, coins sdk.Coins) (res sdk.Coins) {
 	numCoins := 1 + r.Intn(Min(coins.Len(), 1))
 	denomIndices := r.Perm(numCoins)
 	for i := 0; i < numCoins; i++ {
 		denom := coins[denomIndices[i]].Denom
-		amt, _ := simtypes.RandPositiveInt(r, coins[i].Amount)
+		amt, err := simtypes.RandPositiveInt(r, coins[i].Amount)
+		if err != nil {
+			panic(err)
+		}
 		res = append(res, sdk.Coin{Denom: denom, Amount: amt})
 	}
-
 	return
 }
 
+// genQueryCondition returns a single lockup QueryCondition, which is generated from a single coin randomly selected from the provided coin array
 func genQueryCondition(r *rand.Rand, blocktime time.Time, coins sdk.Coins, durations []time.Duration) lockuptypes.QueryCondition {
-	// TODO: reset to 2 after postlaunch, only allow duration based query type on postlaunch
-	// lockQueryType := r.Intn(2)
 	lockQueryType := 0
 	denom := coins[r.Intn(len(coins))].Denom
 	// TODO: for postlaunch, only specific lock durations are allowed
 	// durationSecs := r.Intn(1*60*60*24*7) + 1*60*60 // range of 1 week, min 1 hour
 	// duration := time.Duration(durationSecs) * time.Second
+	// TODO: delete this comment if the below code satisfies the above request
+	lockableDurations := types.DefaultGenesis().LockableDurations
+	durationOk := false
+	for _, duration := range durations {
+		for _, lockableduration := range lockableDurations {
+			if duration == lockableduration {
+				durationOk = true
+				break
+			}
+		}
+		if !durationOk {
+			panic(fmt.Errorf("invalid duration: %d", duration))
+		}
+	}
 	durationIndex := r.Intn(len(durations))
 	duration := durations[durationIndex]
 	timestampSecs := r.Intn(1 * 60 * 60 * 24 * 7) // range of 1 week
```

Comment:
Does the added code satisfy the request to allow only lockable durations?

Reply:
Why are we doing this? The only caller seems to be giving types.DefaultGenesis().LockableDurations for durations anyways? Since this is a test with one use case, do we need this extra validation?

Reply:
ah maybe i misunderstood the TODO of "for postlaunch, only specific lock durations are allowed", I think you are right